tools/clang/test/HLSLFileCheck/hlsl/objects/CbufferLegacy/vec_indexing_with_offset.hlsl

```diff
@@ -0,0 +1,23 @@
+// RUN: %dxc -E main -T ps_6_0 %s | FileCheck %s
```

Comment:
`GenerateCBLoadLegacy` has some code specific to 16-bit types and channels. Maybe it would do well to include an `enable-16bit-types` run? Perhaps with -DTYPE to distinguish them?